packages/back-end/src/util/integrations.ts

```diff
@@ -0,0 +1,109 @@
+import { DataSourceType } from "../../types/datasource";
+import {
+  InformationSchema,
+  RawInformationSchema,
+  Schema,
+  Table,
+} from "../types/Integration";
+
+type RowType = {
+  table_catalog: string;
+  table_schema?: string;
+  table_name?: string;
+  column_name?: string;
+};
+
+function getPath(dataSource: DataSourceType, path: RowType): string {
+  const pathArray = Object.values(path);
+  let returnValue = pathArray.join(".").toLocaleLowerCase();
+  switch (dataSource) {
+    // MySQL only supports path's that go two levels deep. E.G. If the full path is database.schema.table.column, it only supports table.column.
+    case "mysql":
+      if (pathArray.length > 2) {
+        returnValue = pathArray.slice(-2).join(".").toLocaleLowerCase();
```

Comment:
This logic isn't quite right.  In MySQL, the `table_catalog` is always equal to the hard-coded string `"def"`.  So to make the path, you just ignore the catalog and then join everything else the same way as the other data source types.